### PR TITLE
fix: Avoid aliasing known CTEs

### PR DESCRIPTION
**Pull request by snapiri**
I now also add the Common Table Expression (CTE) names to the known tables, so they will not be re-added (thus cross-joined with self).

The following snippet (from the issue) was used to validate the change:
```
    from sqlalchemy import create_engine, MetaData, select, func as F, Table, Column, Integer, ARRAY

    metadata = MetaData()

    table = Table(
        "table1",
        metadata,
        Column("foo", Integer),
        Column("bars", ARRAY(Integer))
    )
    engine = create_engine("bigquery://", future=True)

    selectable = sql.select(table.c).select_from(table).cte('cte')
    q = sql.select([selectable.c.foo, sql.func.unnest(selectable.c.bars).column_valued('unnest_bars')]).select_from(selectable)
    print(q.compile(engine))
```

---

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [X] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-sqlalchemy/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [X] Ensure the tests and linter pass
- [X] Code coverage does not decrease (if any source code was changed)
- [X] Appropriate docs were updated (if necessary)

Related #368

### CODE REVIEW

**Comment by tswast** [comment]
Kokoro failure looks like https://github.com/googleapis/python-bigquery-sqlalchemy/issues/875

We might need to address that before we can merge.

File: sqlalchemy_bigquery/base.py
Comment (tswast) [inline sqlalchemy_bigquery/base.py:264]: Any possibility you could add a test that was failing before this change? Or at least provide a code example so that one of the package maintainers could add such a test.

Comment (snapiri) [reply sqlalchemy_bigquery/base.py:264]: Unit tests added

**Review by tswast — APPROVED** [review]

**Pull request merged**
